powersimdata/design/generation/curtailment.py

```diff
@@ -0,0 +1,124 @@
+import pandas as pd
+
+from powersimdata.scenario.scenario import Scenario
+
+default_pmin_dict = {
+    "coal": None,
+    "dfo": 0,
+    "geothermal": 0.95,
+    "hydro": None,
+    "ng": 0,
+    "nuclear": 0.95,
+    "other": 0,
+    "solar": 0,
+    "wind": 0,
+    "wind_offshore": 0,
+}
+profile_methods = {
+    "hydro": "get_hydro",
+    "solar": "get_solar",
+    "wind": "get_wind",
+    "wind_offshore": "get_wind",
+}
+
+
+def temporal_curtailment(
+    scenario, pmin_by_type=None, pmin_by_id=None, curtailable={"solar", "wind"}
+):
+    """Calculate the minimum share of potential renewable energy that will be curtailed
+        due to supply/demand mismatch, assuming no storage is present.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
+    :param dict/pandas.Series pmin_by_type: Mapping of types to Pmin assumptions. Values
```

Comment:
it seems that this parameter will always be a `dict`

Reply:
The fact that the values of the `pmin_by_type` dictionary can be a `dict` or `pandas.Series` should be mentioned in the description.

Reply:
`pmin_by_type` can be either a dict or a Series, the values should all be float either way.

Reply:
Sorry I got confused with `check_dicts` that encapsulate `pmin_by_type` and `pmin_by_id` in a `dict`